src/ddscxx/src/org/eclipse/cyclonedds/core/cond/StatusConditionDelegate.cpp

```diff
@@ -53,7 +53,15 @@ org::eclipse::cyclonedds::core::cond::StatusConditionDelegate::close()
 {
     /* todo: cannot close this->ddsc_entity because this is the parent EntityDelegate;
         find better solution to get correct feedback when closed StatusCondition is used */
-    this->ddsc_entity = 0;
+
+    // detach the condition from any waitset
+    this->check();
+    org::eclipse::cyclonedds::core::ScopedObjectLock scopedLock(*this);
+    ConditionDelegate::detach_from_waitset(ddsc_entity);
+    // set the entity to 0
+    this->ddsc_entity = DDS_HANDLE_NIL;
```

Comment:
Can't really close the status condition because of the reason above, so currently it detaches the condition from any waitset but doesn't close the entity.

However, since the entity handle is reset to nil, it's anyway not usable. I think I didn't make this any worse if I didn't improve the behavior.